pkg/clusteragent/admission/mutate/autoinstrumentation/auto_instrumentation.go

```diff
@@ -568,41 +473,6 @@ func (w *Webhook) extractLibrariesFromAnnotations(pod *corev1.Pod) []libInfo {
 	return libList
 }
 
-// ShouldInject returns true if Admission Controller should inject standard tags, APM configs and APM libraries
-func ShouldInject(pod *corev1.Pod, wmeta workloadmeta.Component) bool {
-	// If a pod explicitly sets the label admission.datadoghq.com/enabled, make a decision based on its value
-	if val, found := pod.GetLabels()[common.EnabledLabelKey]; found {
-		switch val {
-		case "true":
-			return true
-		case "false":
-			return false
-		default:
-			log.Warnf("Invalid label value '%s=%s' on pod %s should be either 'true' or 'false', ignoring it", common.EnabledLabelKey, val, mutatecommon.PodString(pod))
-		}
-	}
-
-	apmWebhook, err := GetWebhook(wmeta)
-	if err != nil {
-		return config.Datadog().GetBool("admission_controller.mutate_unlabelled")
-	}
-
-	return apmWebhook.isEnabledInNamespace(pod.Namespace) || config.Datadog().GetBool("admission_controller.mutate_unlabelled")
-}
-
-// isEnabledInNamespace indicates if Single Step Instrumentation is enabled for
-// the namespace in the cluster
-func (w *Webhook) isEnabledInNamespace(namespace string) bool {
```

Comment:
This is now a public package function since this is what was used in `ShouldInject` that created the dependency on the this webhook for _other ones_. The `filter` is a global variable instead of the webhook.